### PR TITLE
Add documentation for PgVector Knowledge Base

### PR DESCRIPTION
**Pull request by tswarmerdam-mx**

### CODE REVIEW

File: content/en/docs/appstore/modules/openai/rag-example-implementation.md [hunk 1]
```diff
@@ -75,10 +80,10 @@ This section lists some general key points that apply regardless of which archit
...
+* For RAG, you need a storage space for high-dimensional embedding vectors outside of your normal Mendix app database. Typically, this is a remote vector database. In order to connect to it, the OpenAI showcase application uses the Mendix database connector. See [Vector Database Setup](/appstore/modules/pgvector-knowledge-base/vector-database-setup/) for more details.
 
-* The OpenAI showcase application relies on a PostgreSQL database with the pgvector extension included. Even if you choose to use a different vector database, the database queries and statements, which you will have to perform, should cover at least the following:
-    * Include the vector extension if applicable (create extension).
+* The OpenAI showcase application relies on a PostgreSQL database with the pgvector extension included. In such a setup you can similarly rely on the PgVector Knowledge Base module to take care of creating and executing the right queries. If you choose to use a different type of vector database, the database queries and statements, which you will have to perform, should cover at least the following:
```
Comment (rik-bos) [inline content/en/docs/appstore/modules/openai/rag-example-implementation.md]: Can we add another link to the documentation of the PgVector KB module?

File: content/en/docs/appstore/modules/openai/rag-example-implementation.md [hunk 2]
```diff
@@ -88,7 +93,7 @@ If you would like to build your own RAG setup, feel free to learn from the OpenA
     
     * Remove individual records (delete) or tables (drop table).
 
-{{% alert color="info" %}}Example queries in the form of SQL statements are available for inspiration in the source code of the OpenAI showcase application.{{% /alert %}}
+{{% alert color="info" %}}Example queries in the form of SQL statements are available for inspiration in the source code of the PgVector Knowledge Base module which comes automatically with OpenAI showcase application.{{% /alert %}}
```
Comment (rik-bos) [inline content/en/docs/appstore/modules/openai/rag-example-implementation.md]: Idem

File: content/en/docs/appstore/modules/pgvector-knowledge-base/vector-database-setup.md [hunk 3]
```diff
@@ -76,15 +78,17 @@ You can use the values in the steps below for experimental purposes:
 
 If no action is taken, resources in AWS will stay around indefinitely. Make sure to think about deleting the resources when you are done experimenting. When using services from AWS, you are responsible for having the necessary resources and deleting the ones that are no longer needed, to prevent from being charged more than is required. This is especially relevant the moment resources fall outside of the free-tier after a certain time.
 
-## 4 Configuring the Database Connection Details in the Showcase Application {#configure-database-connection}
+## 4 Configuring the Database Connection Details in your application {#configure-database-connection}
 
-1. Download, run, and login to the [OpenAI Showcase App](https://marketplace.mendix.com/link/component/220475).
+1. Add the [PgVector Knowledge Base module](https://marketplace.mendix.com/link/component/2888) to your Mendix app and set it up correctly, see [PgVector Knowledge Base](/appstore/modules/pgvector-knowledge-base/).
```
Comment (rik-bos) [inline content/en/docs/appstore/modules/pgvector-knowledge-base/vector-database-setup.md]: "and it's dependencies"

File: content/en/docs/appstore/modules/pgvector-knowledge-base/_index.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+Also without invoking LLMs directly with the retrieved information, the similarity search logic from the retrieval operation can be leveraged in combination with embedding models to create a semantic search in a Mendix app. This can be used for fuzzy search capabilities, suggestions or simple recommendation systems.
+
+### 1.2 Features {#features}
+
+With the current version, Mendix supports inserting data chunks with their vectors into a knowledge base (population), and selecting those records from that moment onwards (retrieval). Apart from cosine similarity search, which is executed based on the vector only, custom filtering is possible using key/value labelling to support an additional traditional search component.
+
+### 1.3 Limitations {#limitations}
+
+The current scope of the module is focused around (re)populating knowledge bases as a whole in one single operation. Deleting, adding or updating individual knowledge base items is not yet supported. Furthermore, connections that require drivers other than JDBC are not supported by this module.
```
Comment (rik-bos) [inline content/en/docs/appstore/modules/pgvector-knowledge-base/_index.md]: What do you mean with the last statement?

File: content/en/docs/appstore/modules/pgvector-knowledge-base/_index.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+### 3.1 General Configuration {#general-configuration}
+
+1. Add the module role **PgVectorKnowledgeBase.Administrator** to your Administrator user role in the security settings of your app. Optionally, map **PgVectorKnowledgeBase.User** to any user roles that need read access directly on retrieved entities, [Chunk](#4121-chunk-chunk) and [Label](#4122-label-label).
+2. Add the **DatabaseConfiguration_Overview** page (**USE_ME > Configuration**) to your navigation or add the **Snippet_DatabaseConfigurations** to a page that is already part of your navigation. 
+3. Now you can set up your database configuration(s) at runtime. See [Configuring the database connection details](/appstore/modules/pgvector-knowledge-base/vector-database-setup/#configure-database-connection) for more information.
+
+### 3.2 General operations {#general-operations-configuration}
```
Comment (rik-bos) [inline content/en/docs/appstore/modules/pgvector-knowledge-base/_index.md]: Can we add a typical patterns section right under this section to show how you should chain the operations? E.g. (top of my head)

## for population:
* Create ChunkList
* Create LabelList for chunk
* Create Label (0...n times)
* Create Chunk
* Repopulate

## Retrieve
* Create labellist
* Create label
* Retrieve top k

**Review by rik-bos — APPROVED** [review]

File: content/en/docs/appstore/modules/openai/rag-example-implementation.md (hunk 1 above)
Comment (tswarmerdam-mx) [reply content/en/docs/appstore/modules/openai/rag-example-implementation.md]: done

File: content/en/docs/appstore/modules/openai/rag-example-implementation.md (hunk 2 above)
Comment (tswarmerdam-mx) [reply content/en/docs/appstore/modules/openai/rag-example-implementation.md]: done

File: content/en/docs/appstore/modules/pgvector-knowledge-base/_index.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+
+With the current version, Mendix supports inserting data chunks with their vectors into a knowledge base (population), and selecting those records from that moment onwards (retrieval). Apart from cosine similarity search, which is executed based on the vector only, custom filtering is possible using key/value labelling to support an additional traditional search component.
+
+### 1.3 Limitations {#limitations}
+
+The current scope of the module is focused around (re)populating knowledge bases as a whole in one single operation. Deleting, adding, or updating individual knowledge base items is not yet supported. 
+
+### 1.4 Prerequisites {#prerequisites}
+
+You should have access to your own (remote) postgreSQL database server with the [pgvector](https://github.com/pgvector/pgvector) extension installed. For more information, see [Setting up a Vector Database](/appstore/modules/pgvector-knowledge-base/vector-database-setup/).
```
Comment (rik-bos) [inline content/en/docs/appstore/modules/pgvector-knowledge-base/_index.md:39]: Can we explicitly add the following?

> Note: this module cannot be used with the Mendix Cloud app database, but only works with your own database server or Amazon RDS.

**Pull request merged**